client/src/c_console.cpp

```diff
@@ -105,8 +109,19 @@ static struct NotifyText
 	byte text[256];
 } NotifyStrings[NUMNOTIFIES];
 
-#define PRINTLEVELS 5
-int PrintColors[PRINTLEVELS+1] = { CR_RED, CR_GOLD, CR_GRAY, CR_GREEN, CR_GREEN, CR_GOLD };
+// Default Printlevel
+#define PRINTLEVELS 8 //(5 + 3)
+int PrintColors[PRINTLEVELS] = 
```

Comment:
I see that you removed the `+1` here.  I think this causes a problem elsewhere in the program.  Apparently, the implementation of the cvar `msgmidcolor` uses `PRINTLEVELS` as its default value, which means that the function `setmsgcolor` tries to pull an out-of-bounds index into the `PrintColors` array.  I'm not really sure that having `PRINTLEVELS` act as a "zero value" was the best design idea, so see if you can figure out a more elegant way of solving this issue.